### PR TITLE
Only allow standard ascii in channel names

### PR DESCRIPTION
**Pull request by pfrazee**
Some people had been sneaking gibbering into the channel names. I figure it's better to just normalize them, for now.

### CODE REVIEW

**Comment by pfraze** [comment]
@dominictarr any objections to this?

**Comment by kyrias** [comment]
Being non-American/non-English, there are quite a few instances I can think of where limiting it to ASCII-only is sort of annoying.  But I do agree that it probably shouldn't be completely free-for-all.

File: api/processor.js [hunk 1]
```diff
@@ -404,9 +407,11 @@ function nonEmptyStr (str) {
   }
 
 // allow A-z0-9._-, dont allow a trailing .
-var badNameCharsRegex = /[^A-z0-9\._-]/g
-function makeNameSafe (str) {
-  str = str.replace(badNameCharsRegex, '_')
+var badNameCharsRegex = /[^A-Za-z0-9\._-]/g
```
Comment (ralphtheninja) [inline api/processor.js]: Any good way of refactoring this regexp into a function? Seems it's used at five places.

Comment (pfraze) [inline api/processor.js]: yeah I've been meaning to move it into patchkit-util

Comment (ralphtheninja) [inline api/processor.js]: > yeah I've been meaning to move it into patchkit-util

:+1:

**Pull request closed without merging**